crates/oxc_ast/src/ast_builder_impl.rs

```diff
@@ -112,6 +112,20 @@ impl<'a> AstBuilder<'a> {
         mem::replace(decl, empty_decl)
     }
 
+    #[inline]
+    pub fn move_variable_declaration(
+        self,
+        decl: &mut VariableDeclaration<'a>,
+    ) -> VariableDeclaration<'a> {
+        let empty_decl = self.variable_declaration(
+            Span::default(),
+            VariableDeclarationKind::Var,
+            self.vec(),
+            false,
+        );
+        mem::replace(decl, empty_decl)
+    }
```

Comment:
Will the `var` raise some effects on `KeepVar`?

Reply:
Not sure what you mean, but you can review the added test cases.

Reply:
Do you mean this case?

```JavaScript
for(let i = 0; i < 10; i++) {
    var a = 0;
}
```

In this case, we should not count the variable `a` because it is in block scope.

Reply:
<img width="462" alt="Screenshot 2024-10-14 at 13 12 59" src="https://github.com/user-attachments/assets/9ff07646-26ea-4491-ac51-4c33bc9a7899">

why do we drop `a` ?

Reply:
Well, I got misunderstood. You can ignore this review :joy: